src/speed_lookup_bindings.cpp

```diff
@@ -177,7 +177,22 @@ NAN_METHOD(SpeedLookup::getRouteSpeeds)
             : Base(callback), datamap{datamap_}, nodeIds(std::move(nodeIds))
         {
         }
-        void Execute() override { result_annotations = datamap->getValues(nodeIds); }
+
+        void Execute() override { 
+
+            if (datamap)
+            {
+                result_annotations = datamap->getValues(nodeIds); 
+            }
+            else
+            {
+                for (std::size_t index = 0; index < nodeIds.size(); index++)
+                {
+                    result_annotations.emplace_back(INVALID_SPEED);
+                }
```

Comment:
The more C++-y way to do this would be:

```
result_annotations.resize(nodeIDs.size() -1);
std::fill(result_annotations.begin(), 
          result_annotations.end(), 
          INVALID_SPEED);
```

This is because `std::fill` might be able to fill the array more efficiently than a simple loop (for example, by the compiler taking advantage of https://en.wikipedia.org/wiki/SIMD instructions in the CPU).

The main idea here is that it's more obvious what's happening if you say `std::fill()` than if you write a loop - the compiler (and other readers) can recognize that and make optimizations you might not have thought of.

To use `std::fill`, you'll need to add `#include <algorithm>` at the top of this file.

C++ has a whole set of these kinds of helper functions: http://en.cppreference.com/w/cpp/algorithm